src/CanvasRenderingContext2d.cc

```diff
@@ -292,6 +292,58 @@ Context2d::restorePath() {
   cairo_path_destroy(_path);
 }
 
+/*
+ * Create temporary surface for gradient or pattern transparency
+ */
+cairo_pattern_t*
+create_transparent_gradient(cairo_pattern_t *source, float alpha) {
+  double x0;
+  double y0;
+  double x1;
+  double y1;
+  double r0;
+  double r1;
+  int count;
+  int i;
+  double offset;
+  double r;
+  double g;
+  double b;
+  double a;
+  cairo_pattern_t *newGradient;
+  cairo_pattern_type_t type = cairo_pattern_get_type(source);
+  cairo_pattern_get_color_stop_count(source, &count);
+  if (type == 2) {
```

Comment:
can you use the `cairo_pattern_type_t` constants instead? https://developer.gnome.org/cairo/stable/cairo-cairo-pattern-t.html#cairo-pattern-type-t